test/Microsoft.Health.Fhir.Shared.Tests.Integration/Persistence/SearchParameterOptimisticConcurrencyIntegrationTests.cs

```diff
@@ -0,0 +1,494 @@
+
+                // Assert
+                Assert.Equal(SearchParameterStatus.Enabled, result.Status);
+                Assert.True(result.IsPartiallySupported);
+                Assert.True(result.LastUpdated > createdStatus.LastUpdated, "LastUpdated should change after update");
+            }
+            finally
+            {
+                // Cleanup
+                await _testHelper.DeleteSearchParameterStatusAsync(testUri);
+            }
+        }
+
+        [SkippableFact]
+        public async Task GivenExistingSearchParameterStatus_WhenUpdatingWithIncorrectLastUpdated_ThenEventuallySucceedsWithRetry()
+        {
+            // Skip if not SQL Server or schema version < 94
+            Skip.If(!IsSqlServerWithOptimisticConcurrency(), "Schema version 94+ required for optimistic concurrency");
+
+            // Arrange
+            var testUri = $"http://test.com/SearchParameter/ConcurrencyTest_{Guid.NewGuid()}";
+            var initialStatus = new ResourceSearchParameterStatus
+            {
+                Uri = new Uri(testUri),
+                Status = SearchParameterStatus.Disabled,
+                IsPartiallySupported = false,
+            };
+
+            try
+            {
+                // Create initial status
+                await _fixture.SearchParameterStatusDataStore.UpsertStatuses(new[] { initialStatus }, CancellationToken.None);
+
+                // Get the created status with its LastUpdated
+                var allStatuses = await _fixture.SearchParameterStatusDataStore.GetSearchParameterStatuses(CancellationToken.None);
+                var createdStatus = allStatuses.First(s => s.Uri.ToString() == testUri);
+
+                // Make an intermediate update to change the LastUpdated
+                var intermediateUpdate = new ResourceSearchParameterStatus
+                {
+                    Uri = createdStatus.Uri,
+                    Status = SearchParameterStatus.Enabled,
+                    IsPartiallySupported = false,
+                    LastUpdated = createdStatus.LastUpdated,
+                };
+                await _fixture.SearchParameterStatusDataStore.UpsertStatuses(new[] { intermediateUpdate }, CancellationToken.None);
+
+                // Now try to update with the stale LastUpdated
+                // The retry mechanism should detect the conflict, refresh the LastUpdated, and succeed
+                var staleUpdate = new ResourceSearchParameterStatus
+                {
+                    Uri = createdStatus.Uri,
+                    Status = SearchParameterStatus.Supported,
+                    IsPartiallySupported = true,
+                    LastUpdated = createdStatus.LastUpdated, // This is now stale
+                };
+
+                // Act - This should succeed due to retry mechanism
+                await _fixture.SearchParameterStatusDataStore.UpsertStatuses(new[] { staleUpdate }, CancellationToken.None);
```

Comment:
Is there any way to test that the retry actually happened? This is showing that the upsert worked, but it doesn't show why it is working.